source/recovery.rst

```diff
@@ -42,3 +42,122 @@ Applying this option forces the cluster to start. However, there may exist data
 inconsistency in the cluster, and several last transactions may be lost. 
 If such data loss is undesirable, experienced users may choose the more advanced
 manual method described in the next chapter.
+
+.. _recovery-object-surgery:
+
+Object Surgery Crash Recovery method
+------------------------------------
+
+.. warning:: This method is intended for advanced users only!
+
+This method involves the following steps:
+* swap the original PXC image with the debug image, which does not reboot after
+  the crash, and force all Pods to run it,
+* find the Pod with the most recent PXC data, run recovery on it, start
+  ``mysqld``, and allow the cluster to be restarted,
+* revert all temporary substitutions.
+
+Let's assume that a full crash did occur for the cluster named ``cluster1``,
+which is based on three PXC Pods.
+
+.. note:: The following commands are written for PXC 8.0. The same steps are
+   also for PXC 5.7 unless specifically indicated otherwise.
+
+1. Change the normal PXC image inside the cluster object to the debug image:
+
+   .. code-block:: bash
+
+      $ kubectl patch pxc cluster1 --type="merge" -p '{"spec":{"pxc":{"image":"perconalab/percona-xtradb-cluster-operator:master-pxc8.0-debug"}}}'
+
+   .. note:: For PXC 5.7 this command should be as follows:
+
+      .. code-block:: bash
+
+         $ kubectl patch pxc cluster1 --type="merge" -p '{"spec":{"pxc":{"image":"perconalab/percona-xtradb-cluster-operator:master-pxc5.7-debug"}}}'
+
+2.  Restart all Pods:
+
+   .. code-block:: bash
+
+      $ $ for i in $(seq 0 $(($(kubectl get pxc cluster1 -o jsonpath='{.spec.pxc.size}')-1))); do kubectl delete pod cluster1-pxc-$i --force --grace-period=0; done
+
+3. Wait until the Pod ``0`` is ready, and execute the following code (it is
```

Comment:
In the updated doc we have restarted not *Pod 0*, but *all Pods* on the previous step. Now we wait until "Pod 0" is ready and execute some code (perhaps for all pods). Shouldn't we write here "Wait until all Pods are ready"?

Reply:
no, it is ok